core/src/main/java/org/apache/accumulo/core/iterators/system/SourceSwitchingIterator.java

```diff
@@ -162,6 +162,14 @@ private void readNext(boolean initialSeek) throws IOException {
   private boolean switchSource() throws IOException {
     if (!source.isCurrent()) {
       source = source.getNewDataSource();
+      // if our source actually changed, then attempt to close the previous iterator
+      try {
+        if (iter != null) {
+          iter.close();
+        }
+      } catch (Exception e) {
+        throw new IOException(e);
```

Comment:
This is a private method. Do we need to wrap exceptions here, or should we handle it in the caller? It seems we might be converting them to IOExceptions here, only to have them converted back to RuntimeExceptions later.